src/test/run-pass/associated-const-type-parameters.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(associated_consts)]
+
+trait Foo {
+    const X: u32;
+    fn get_x() -> u32 {
+       Self::X
+    }
+}
+
+struct Abc;
+impl Foo for Abc {
+    const X: u32 = 11;
+}
+
+struct Def;
+impl Foo for Def {
+    const X: u32 = 97;
+}
+
+fn add<A: Foo, B: Foo>() -> u32 {
+    A::X + B::X
+}
+
+fn main() {
+    println!("{}", Abc::X);
+    println!("{}", Def::X);
+    println!("{}", Abc::get_x());
+    println!("{}", Def::get_x());
+    println!("{}", add::<Abc, Def>());
+    assert_eq!(108, add::<Abc, Def>());
```

Comment:
also, can you change this from `add` to `sub`, so that we know that the order is correct? that way, we can also test the result of `sub::<Def, Abc>()`, for example.